components/script/dom/urlhelper.rs

```diff
@@ -149,6 +153,7 @@ impl UrlHelper {
     }
 
     pub fn SetSearch(url: &mut Url, value: USVString) {
+        url.query = Some(String::new());
```

Comment:
I’m rebasing #9840 which basically rewrites this file. Why was this change (and others in this file) necessary?